src/js/jsx/sections/libraries/LibraryList.jsx

```diff
@@ -96,24 +160,203 @@ define(function (require, exports, module) {
             return options;
         },
 
-        render: function () {
+        /**
+         * Handle change of current library's name. The new name is not committed yet.
+         *
+         * @private
+         * @param  {SyntheticEvent} event
+         * @param  {string} newName
+         */
+        _handleChangeName: function (event, newName) {
+            this.setState({
+                newLibraryName: newName
+            });
+        },
+
+        /**
+         * Invoke libraries action to change the current library's name.
+         *
+         * @private
+         */
+        _handleRename: function () {
+            if (this.state.newLibraryName.length !== 0) {
+                this.getFlux().actions.libraries.renameLibrary(this.props.selected.id, this.state.newLibraryName);
+                this.setState({ command: null });
+            }
+        },
+
+        /**
+         * Invoke libraries action to create a library
+         *
+         * @private
+         */
+        _handleCreate: function () {
+            if (this.state.newLibraryName.length !== 0) {
+                this.getFlux().actions.libraries.createLibrary(this.state.newLibraryName);
+                this.setState({ command: null });
+            }
+        },
+
+        /**
+         * Invoke libraries action to delete the current library
+         *
+         * @private
+         */
+        _handleDelete: function () {
+            this.getFlux().actions.libraries.removeLibrary(this.props.selected.id);
+            this.setState({ command: null });
+        },
+
+        /**
+         * Cancel the current library command (create / rename / delete).
+         *
+         * @private
+         */
+        _handleCancel: function () {
+            this.setState({ command: null });
+        },
+
+        /**
+         * Render confirmation dialog for deleting the current library.
+         *
+         * @private
+         * @return {?ReactComponent}
+         */
+        _renderDeleteConfirmationDialog: function () {
+            if (this.state.command !== _DELETE_LIBRARY) {
+                return null;
+            }
+
+            var collaborationMapBody = {};
```

Comment:
Unfortunately Chrome does not support using variable as property name in {} syntax and I want to avoid repeating the category strings here. Note that this is one of the features in ECMAScript 6, which will allow us to write something like these:

```js
var collaborationMapBody = {
  [_OUTGOING_LIBRARY]: ...,
  [_INCOMING_LIBRARY]: ...,
}
```

Given that `collaborationMapBody` does not look like a constant, and putting it in top level will cache the translation strings (although we cannot change language in DS), this might be a better place for it.